src/Fixer/Import/GroupImportFixer.php

```diff
@@ -0,0 +1,256 @@
+use PhpCsFixer\Tokenizer\Analyzer\Analysis\NamespaceUseAnalysis;
+use PhpCsFixer\Tokenizer\Analyzer\NamespaceUsesAnalyzer;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Vladimir Kuprienko <vldmr.kuprienko@gmail.com>
+ */
+final class GroupImportFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'There MUST be group use for the same namespaces.',
+            [
+                new VersionSpecificCodeSample(
+                    "<?php\nuse Foo\\Bar;\nuse Foo\\Baz;\n",
+                    new VersionSpecification(70000)
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return \PHP_VERSION_ID >= 70000 && $tokens->isTokenKindFound(T_USE);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $useWithSameNamespaces = $this->getSameNamespaces($tokens);
+
+        if (empty($useWithSameNamespaces)) {
+            return;
+        }
+
+        $this->removeSingleUseStatements($useWithSameNamespaces, $tokens);
+        $this->addGroupUseStatements($useWithSameNamespaces, $tokens);
+    }
+
+    /**
+     * Gets namespace use analyzers with same namespaces.
+     *
+     * @param Tokens $tokens
+     *
+     * @return array|NamespaceUseAnalysis[]
+     */
+    private function getSameNamespaces(Tokens $tokens)
+    {
+        $useDeclarations = (new NamespaceUsesAnalyzer())->getDeclarationsFromTokens($tokens);
+
+        if (0 === \count($useDeclarations)) {
+            return [];
+        }
+
+        $allNamespaces = array_map(
+            function ($useDeclaration) {
+                return $this->getNamespaceNameWithSlash($useDeclaration);
+            },
+            $useDeclarations
+        );
+
+        $sameNamespaces = array_filter(array_count_values($allNamespaces), function ($count) {
+            return $count > 1;
+        });
+        $sameNamespaces = array_keys($sameNamespaces);
+
+        $sameNamespaceAnalysis = array_filter($useDeclarations, function ($useDeclaration) use ($sameNamespaces) {
+            $namespaceName = $this->getNamespaceNameWithSlash($useDeclaration);
+
+            return \in_array($namespaceName, $sameNamespaces, true);
+        });
+
+        sort($sameNamespaceAnalysis);
+
+        return $sameNamespaceAnalysis;
+    }
+
+    /**
+     * @param NamespaceUseAnalysis[] $statements
+     * @param Tokens                 $tokens
+     */
+    private function removeSingleUseStatements($statements, Tokens $tokens)
+    {
+        foreach ($statements as $useDeclaration) {
+            $index = $useDeclaration->getStartIndex();
+            $endIndex = $useDeclaration->getEndIndex() + 2; // ;\n
```

Comment:
Added test case.

Reply:
Is it critical to have 

```php
use Foo\{Bar, Baz};\DontTouch::me();
```

or it can be

```php
use Foo\{Bar, Baz};
\DontTouch::me();
```
?

Reply:
It would be better but it's not critical so we can merge with the added newline and try to improve the fixer afterwards.

Reply:
Fixed, but with adding newline.